source/opt/ir_context.h

```diff
@@ -576,6 +576,24 @@ class IRContext {
     return GetFunction(inst->result_id());
   }
 
+  // Updates DebugInlinedAt of |inst| and analyzes use information
+  // of DebugInlinedAt.
+  void UpdateDebugInlinedAt(Instruction* inst, uint32_t dbg_inlined_at) {
+    inst->UpdateDebugInlinedAt(dbg_inlined_at);
+    if (AreAnalysesValid(kAnalysisDebugInfo)) {
+      get_debug_info_mgr()->AnalyzeDebugInst(inst);
+    }
+  }
+
+  // Updates DebugScope and line of |inst| and analyzes use information
```

Comment:
You can update the analysis in `Instruction::UpdateDebugInfoFrom`.  The instruction class has access to the context.

Reply:
done